src/rt/argv_windows.d

```diff
@@ -0,0 +1,237 @@
+module rt.argv_windows;
+
+version (Windows):
+
+import core.stdc.wchar_ /+: wcslen+/;
+import core.sys.windows.winbase /+: LocalAlloc+/;
+import core.sys.windows.winnt /+: LPWSTR, LPCWSTR+/;
+
+/++
+Splits a command line into argc/argv lists, using the VC7 parsing rules.
+This functions interface mimics the `CommandLineToArgvW` api.
+If function fails, returns NULL.
+If function suceeds, call `LocalFree(HLOCAL)` on return pointer when done.
+NOTE Implementation-wise, once every few years it would be a good idea to
+compare this code with the .NET Runtime's `SegmentCommandLine` method,
+which is in `src/coreclr/src/utilcode/util.cpp`.
+Date: Feb 4, 2020
++/
+LPWSTR* commandLineToArgv(LPCWSTR lpCmdLine, int* pNumArgs) nothrow @nogc
+{
+    /+
+    The MIT License (MIT)
+
+    Copyright (c) .NET Foundation and Contributors
+
+    All rights reserved.
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in all
+    copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+    +/
+
+    *pNumArgs = 0;
+
+    int nch = cast(int) wcslen(lpCmdLine);
+
+    // Calculate the worstcase storage requirement. (One pointer for
+    // each argument, plus storage for the arguments themselves.)
+    int cbAlloc = (nch + 1) * (cast(int) LPWSTR.sizeof) + (nch + 1) * (cast(int) WCHAR.sizeof);
+    LPWSTR pAlloc = cast(wchar*) LocalAlloc(LPTR, cbAlloc);
```

Comment:
It was intended for compatibility with older CommandLineToArgvW which returns a LocalAlloc-ed buffer to avoid further modifications in dmain2.d. But it could be changed to some better solution.